### PR TITLE
ClassCastException during Graal native image and ReST Client

### PR DESCRIPTION
**Pull request by dansiviter**
I've had issues testing this locally. I can replicate the issue and the build gets further with these changes, but ultimately still fails but for another reason.

### CODE REVIEW

**Comment by dansiviter** [comment]
I've updated the manual integration tests to trigger a `int` primitive `@PathParam` on the ReST Client.

File: tests/integration/native-image/mp-1/src/main/java/io/helidon/tests/integration/nativeimage/mp1/Mp1Main.java [hunk 1]
```diff
@@ -191,8 +191,9 @@ private static void testBean(int port, String jwtToken) {
         // + JSON-B
         invoke(collector, "Rest client JSON-B", "json-b", aBean::restClientJsonB);
         // + query params
-        invoke(collector, "Rest client Query param long", "1020", () -> aBean.restClientQuery(1020L));
```
Comment (dansiviter) [inline tests/integration/native-image/mp-1/src/main/java/io/helidon/tests/integration/nativeimage/mp1/Mp1Main.java:194]: I updated the message to reflect they're targeting the wrapper Objects and not the primitive type.

**Comment by m0mus** [comment]
Fixes #2913

**Comment by dansiviter** [comment]
Not sure why it's stating test failures. When I click on it it says `BUILD SUCCESS` in the log.

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**